### ISSUE TITLE
Overlay HTML Problem

### ISSUE THREAD

**Issue by eg271** [mapsplugin/v2.0-demo#47]
First off, great work with V2 - looking good!

I have a question about using this in a Single Page App (same HTML page).  Let's say that I want to load the map once and only once when the app starts, keep it in the background and hide/show as needed for performance reasons.  How do you place overlay elements?

In previous iterations of v2, you could place another blank div as an overlay which didn't have to be a child of the map div itself.  This would have a transparent background and still allow user interaction with the map where child elements (eg. like a custom scale or the tab's bottom menu bar) weren't positioned.

But in the latest version, the plugin seems to block all input even where the blank overlay is invisible and no child is showing.  In iOS, if I remove the overlay div manually from Safari Inspector, I can interact with the map again.

I realise that the map could be placed in a div on the specific map tab but this would mean that it is created and destroyed each time and so take a moment to load.  What's the solution here?  Is there a way to override the plugin detecting the invisible div to let input of the map through as was it's previous behaviour?

**Comment by wf9a5m75** [comment]
Looks like your app is complicated. Please share your project files on github or bitbucket.
How and why?
https://github.com/mapsplugin/cordova-plugin-googlemaps-doc/tree/master/articles/before_post_your_issue#6-provide-your-code

**Comment by eg271** [comment]
I've actually realised the real issue now, so before sharing anything:

The plugin's docs advise to make overlay elements position: fixed.  The problem is that in my case the overlay's parent elements have a transform: translate applied to them which is needed for page transition animations.  Unfortunately though, this makes the position fixed relative to the _parent_ and not the window (because it resets the coordinate system).  This is a known browser issue and an intentional part of the spec: http://meyerweb.com/eric/thoughts/2011/09/12/un-fixing-fixed-elements-with-css-transforms/

Therefore, the solution I implemented was to make a full height transparent container using CSS flex box which covers the entire map and holds any overlay elements.  Except in the most recent V2 this now blocks user interaction even where transparent.

So, the ultimate issue seems to be that the plugin has removed support for transparent overlay elements, useful in situations where position fixed can't be used.

**Comment by wf9a5m75** [comment]
>The plugin's docs advise to make overlay elements position: fixed. 

Where?

**Comment by wf9a5m75** [comment]
I'm not sure you understand this plugin or not, but this plugin displays the google maps view **UNDER** the browser view. The Google Maps view is not an HTML element.

![](https://user-images.githubusercontent.com/167831/28292982-910af668-6b06-11e7-933e-42311a58e64e.png)

So the elements above the map div must be transparent.

**Comment by eg271** [comment]
> The plugin's docs advise to make overlay elements position: fixed.
> > Where?

Change 3: Recognise all HTML elements in https://github.com/mapsplugin/cordova-plugin-googlemaps-doc/blob/master/v2.0.0/whats-new-v2/README.md


Perhaps my explanation wasn't very clear - I was using the term "overlay elements" to refer to HTML elements not anything on the native map.  Anyway, please try this very simple example so hopefully you can understand what I'm referring to:
https://bitbucket.org/Edgraham2/cordova-google-maps-test/src

After adding an app & API key combo in config.xml and installing the plugins as per usual, if you run it in the iOS simulator you should find that you cannot drag the visible part of the map.  However, if you open up Safari Inspector and manually untick the "height: 100%" CSS rule for #overlay, the map becomes responsive to gestures.

Unfortunately, this also pulls the blue #child div off the screen because of the transform: translateX(0) rule.  Oddly, after reticking the height rule, the map can be dragged even through the blue div but this may be to do with the plugin's detection code not firing?

**Comment by wf9a5m75** [comment]
Ah, I see.
Sorry, I forgot to write the description. 
You don't need to do that anymore.

The v2 plugin recognizes the position, size and z-index of all HTML elements automatically.

**Comment by wf9a5m75** [comment]
Please reinstall the plugin `2.0.0-beta2-20170719-1400` or higher version.

**Comment by wf9a5m75** [comment]
And thank you for sharing your repo. I will try it.

**Comment by eg271** [comment]
Same result with 2.0.0-beta2-20170719-1400

**Comment by wf9a5m75** [comment]
Ok, please wait for a while. I'm working for another issue.

**Comment by wf9a5m75** [comment]
I checked your code. Unfortunately, the map view is not HTML element.
So it does not mach with CSS tricks well.

Please use the Google Maps JavaScript API v3.

**Comment by wf9a5m75** [comment]
I double checked your demo code.

You can not the map-canvas div (red) even pure HTML/JavaScript.

```html
<!DOCTYPE html>

<html>
<head>
    <style>
        html{
            height: 100%;
        }

        body {
            font-size:12px;
            height:100%;
            margin:0px;
            padding:0px;
            width:100%;
        }

        #map-canvas{
            position: absolute;
            top:0;
            bottom:0;
            left:0;
            right:0;
            background-color: red;
        }

        #overlay{
            transform:translateX(0);
            height:100%;
            width:100%;
        }
        #child{
            position: fixed;
            background: blue;
            top: 500px;
            left: 0;
            right: 0;
            bottom: 0;
        }

    </style>
<script>
window.addEventListener("load", function() {
  var mapDiv = document.getElementById("map-canvas");
  mapDiv.addEventListener("click", function() {
    alert("clicked");
  });
});
</script>
    </head>
    <body>
        <div id="map-canvas" ></div>
        <div id="overlay">
            <div id="child"></div>
        </div>

    </body>
</html>
```
![screen shot 2017-07-19 at 8 26 38 pm](https://user-images.githubusercontent.com/167831/28399210-950f0676-6cc0-11e7-8bcb-4ceb2885f803.png)

**Comment by wf9a5m75** [comment]
This plugin reproduces the hierarchy of HTML elements.
If the map div is not clicked as hierarchy, you can not click on the map.

**Comment by eg271** [comment]
Ok, if the map _div_ always has to be in the correct place in the DOM hierarchy, I think the solution is to use setDiv.  When a new tab's view is created, apply map.setDiv() to a new element on this tab.  Even when the element is destroyed as the view is torn down, the map still exists in memory and reapplying setDiv the next time the view is shown reinstates it as before.

Have tested this on my app and seems to work.  So, this seems to be best way to go for apps that create/destroy the map div when navigating to the view but not wanting to reload a map each time.

**Comment by wf9a5m75** [comment]
If you remove the map div, the plugin removes the map automatically.
But as the cordova mechanism, the plugin which has been loaded at once, it keeps in the memory until the end of the app.

The maps plugin cleans up itself as much as possible when the map is removed.
